File: src/mappings/vcs.rs
pub fn get(filetype: &str) -> Option<(&str, &str)> {
    let vcs = match filetype {
        "gitcommit" | "gitrebase" => ("default", "Git"),
        "fugitive" | "fugitiveblame" => ("default", "Fugitive"),
        "magit" => ("default", "Magit"),
        "git.nvim" => ("default", "Git.nvim"),
        "lazygit" => ("default", "Lazygit"),
        _ => {
            if filetype.starts_with("Neogit") {
                ("default", "Neogit")
            } else {
                return None;
            }
        }
    };

    Some(vcs)
}

File: lua/cord/mappings/vcs.lua
local M = {}

M.default_icon = 'git'
local mappings = {
  magit = { M.default_icon, 'Magit' },
  gitcommit = { M.default_icon, 'Git' },
  gitrebase = { M.default_icon, 'Git' },
  fugitive = { M.default_icon, 'Fugitive' },
  fugitiveblame = { M.default_icon, 'Fugitive' },
  lazygit = { M.default_icon, 'Lazygit' },
  ['git.nvim'] = { M.default_icon, 'Git.nvim' },
}

M.get = function(filetype)
  if filetype:match '^Neogit' then return { M.default_icon, 'Neogit' } end

  return mappings[filetype]
end

return M
